### ISSUE TITLE
`jnp.linalg.svd`: subnormal singular values are returned as 0

### ISSUE THREAD

**Issue by JamesDLance** [jax-ml/jax#37766]
### Description

Passing a matrix to `jnp.linalg.svd` where one singular value falls in the float32 subnormal range gives `0.0` for that singular value. `numpy.linalg.svd` returns the correct nonzero value.

I noticed this when using JAX for a low-rank approximation task. The matrix has a very small but meaningful component along one dimension — its singular value is around `1e-40`. After SVD, JAX reports that component as exactly zero, so it gets dropped entirely from the reconstruction.

## Reproducer

```python
import numpy as np
import jax.numpy as jnp

tiny = np.finfo(np.float32).tiny

for sv in [np.nextafter(tiny, np.float32(0)), np.float32(1e-40), np.float32(1e-43)]:
    A = np.diag(np.array([1.0, 0.01, float(sv)], dtype=np.float32))
    np_s  = np.linalg.svd(A, compute_uv=False)
    jax_s = np.asarray(jnp.linalg.svd(jnp.array(A), compute_uv=False))
    print(f"sv={sv:.2e}  |  numpy: {np_s}  |  jax: {jax_s}")
```

## Expected output

```
sv=1.18e-38  |  numpy: [1.0000e+00 1.0000e-02 1.1755e-38]  |  jax: [1.0000e+00 1.0000e-02 1.1755e-38]
sv=1.00e-40  |  numpy: [1.e+00 1.e-02 1.e-40]               |  jax: [1.e+00 1.e-02 1.e-40]
sv=9.95e-44  |  numpy: [1.e+00 1.e-02 9.949e-44]            |  jax: [1.e+00 1.e-02 9.949e-44]
```

## Actual output

```
sv=1.18e-38  |  numpy: [1.0000e+00 1.0000e-02 1.1755e-38]  |  jax: [1.   0.01 0.  ]
sv=1.00e-40  |  numpy: [1.e+00 1.e-02 1.e-40]               |  jax: [1.   0.01 0.  ]
sv=9.95e-44  |  numpy: [1.e+00 1.e-02 9.949e-44]            |  jax: [1.   0.01 0.  ]
```

The smallest singular value is always `0.0` in JAX regardless of its actual magnitude.

## Notes

- The two larger singular values (`1.0` and `0.01`) are correct; only the subnormal one is dropped.
- This silently corrupts rank estimation and pseudoinverse computations that rely on the full singular value spectrum.
- The test matrix is diagonal, so the singular values are just the diagonal entries — no numerical error should be present; the loss of precision is entirely due to the subnormal flush.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:     0.10.1.dev20260513+64f6e229b
jaxlib:  0.10.0
numpy:   2.4.4
scipy:   1.17.1
python:  3.12.3
OS:      Linux-6.14.0-37-generic-x86_64-with-glibc2.39
backend: cpu
```

**Comment by jakevdp** [comment]
Thanks for the report! In general JAX (via XLA) will use flush-to-zero semantics for subnormal numbers on some backends (see https://docs.jax.dev/en/latest/notebooks/Common_Gotchas_in_JAX.html#miscellaneous-divergences-from-numpy) so this is behaving as expected.
